src/ui/public/vis/vis.js

```diff
@@ -61,7 +61,7 @@ export function VisProvider(Private, indexPatterns, timefilter, getAppState) {
         indexPatterns: indexPatterns,
         timeFilter: timefilter,
         queryFilter: queryFilter,
-        queryManager: queryManagerFactory(getAppState()),
+        queryManager: queryManagerFactory(getAppState),
```

Comment:
While fixing the spatial filters I noticed `getAppState()` would return undefined here. It seems like the vis is constructed too early to rely on calling it at this point. So I updated the queryManagerFactory to take a getter function instead of the state itself. I think that'll prove to be an improvement in general.